### DISCUSSION TITLE
Azure App Service with container performance

### DISCUSSION THREAD

**Discussion by oysandvik94** [hardkoded/puppeteer-sharp#1995] [General]
Hi,

I'm currently running Puppeteer Sharp on Azure App Service in a linux docker container. On the cheaper tiers it's limited to just a few CPU cores, and especially when limited to 1 CPU core it runs very slow. Here's my code:

```
var args = new List<string>
{
    "--no-sandbox",
    "--disable-dev-shm-usage",
    "--disable-gpu",
    "--disable-setuid-sandbox",
    "--no-first-run",
    "--no-zygote",
    "--single-process"
};

var browser = await Puppeteer.LaunchAsync(new LaunchOptions
{
    Headless = true,
    Args = args.ToArray()
});

await using var page = await browser.NewPageAsync();

await page.SetContentAsync(html);
var pdf = await page.PdfStreamAsync();

```

Both "NewPageAsync()" and "LaunchAsync" methods are quite slow. On a the 1 CPU B tier this takes around 1-3 seconds. Multiple requests causes hanging issues. 

Upgrading to 2 CPU (still B tier) makes it quite a bit better. Around 800 ms, compared to around 100 ms on my own computer, but no hangs. 

Obviously it's not a big deal going in to a production server, but other tools I've used to convert HTML to PDF don't have the same performance issues on dev resources.

I was really just wondering if anyone else has had the same setup on Azure, and if they have any tips?
I've tried to debug the processes on the Azure instance, and I'm just really seeing CPU spiking to 100%, no errors in the chrome logs.

**Comment by iTz1T4Y07** [comment]
Try using azure container app instead of web app
